commands/command_prune.go

```diff
@@ -327,8 +330,10 @@ func pruneTaskGetRetainedAtRef(gitscanner *lfs.GitScanner, ref string, retainCha
 			return
 		}
 
-		retainChan <- p.Oid
-		tracerx.Printf("RETAIN: %v via ref %v", p.Oid, ref)
+		if gitscanner.Filter.Allows(p.Name) {
```

Comment:
I think this is the behavior that you're looking for, but other APIs from the `gitscanner` code do this filtering from within the `lfs` package, such that file- (or pointer-) names that don't pass the filter aren't propagated out to the callbacks.

For consistency: what do you think about moving those checks there?

Reply:
Of course! That was just necessary because I haven't used your changes from #2847 ... stupid me. Now everything is in master and rebased my branch here.